src/Host/Client/Impl/Session/RSessionProvider.cs

```diff
@@ -102,10 +103,9 @@ public Task<RHost> ConnectAsync(string name, IRCallbacks callbacks, string rComm
         }
 
         public Task<bool> TrySwitchBroker(string name, string path = null) {
-            path = path ?? new RInstallation().GetRInstallPath();
-
+            path = path ?? new RInstallation().GetCompatibleEngines().FirstOrDefault()?.InstallPath;
             Uri uri;
-            if (!Uri.TryCreate(path, UriKind.Absolute, out uri)) {
+            if (string.IsNullOrEmpty(path) || !Uri.TryCreate(path, UriKind.Absolute, out uri)) {
```

Comment:
You don't need this verification - `Uri.TryCreate` handles both nulls and empty strings